Mathlib/Algebra/Field/Basic.lean

```diff
@@ -0,0 +1,345 @@
+
+theorem one_div_neg_one_eq_neg_one : (1 : K) / -1 = -1 :=
+  have : -1 * -1 = (1 : K) := by rw [neg_mul_neg, one_mul]
+  Eq.symm (eq_one_div_of_mul_eq_one_right this)
+
+theorem one_div_neg_eq_neg_one_div (a : K) : 1 / -a = -(1 / a) :=
+  calc
+    1 / -a = 1 / (-1 * a) := by rw [neg_eq_neg_one_mul]
+    _ = 1 / a * (1 / -1) := by rw [one_div_mul_one_div_rev]
+    _ = 1 / a * -1 := by rw [one_div_neg_one_eq_neg_one]
+    _ = -(1 / a) := by rw [mul_neg, mul_one]
+
+theorem div_neg_eq_neg_div (a b : K) : b / -a = -(b / a) :=
+  calc
+    b / -a = b * (1 / -a) := by rw [← inv_eq_one_div, division_def]
+    _ = b * -(1 / a) := by rw [one_div_neg_eq_neg_one_div]
+    _ = -(b * (1 / a)) := by rw [neg_mul_eq_mul_neg]
+    _ = -(b / a) := by rw [mul_one_div]
+
+theorem neg_div (a b : K) : -b / a = -(b / a) := by
+  rw [neg_eq_neg_one_mul, mul_div_assoc, ← neg_eq_neg_one_mul]
+
+@[field_simps]
+theorem neg_div' (a b : K) : -(b / a) = -b / a := by simp [neg_div]
+
+theorem neg_div_neg_eq (a b : K) : -a / -b = a / b := by rw [div_neg_eq_neg_div, neg_div, neg_neg]
+
+theorem neg_inv : -a⁻¹ = (-a)⁻¹ := by rw [inv_eq_one_div, inv_eq_one_div, div_neg_eq_neg_div]
+
+theorem div_neg (a : K) : a / -b = -(a / b) := by rw [← div_neg_eq_neg_div]
+
+theorem inv_neg : (-a)⁻¹ = -a⁻¹ := by rw [neg_inv]
+
+end DivisionMonoid
+
+section DivisionRing
+
+variable [DivisionRing K] {a b : K}
+
+@[simp]
+theorem div_neg_self {a : K} (h : a ≠ 0) : a / -a = -1 := by rw [div_neg_eq_neg_div, div_self h]
+
+@[simp]
+theorem neg_div_self {a : K} (h : a ≠ 0) : -a / a = -1 := by rw [neg_div, div_self h]
+
+theorem div_sub_div_same (a b c : K) : a / c - b / c = (a - b) / c := by
+  rw [sub_eq_add_neg, ← neg_div, div_add_div_same, sub_eq_add_neg]
+
+theorem same_sub_div {a b : K} (h : b ≠ 0) : (b - a) / b = 1 - a / b := by
+  simpa only [← @div_self _ _ b h] using (div_sub_div_same b a b).symm
+
+theorem one_sub_div {a b : K} (h : b ≠ 0) : 1 - a / b = (b - a) / b :=
+  (same_sub_div h).symm
+
+theorem div_sub_same {a b : K} (h : b ≠ 0) : (a - b) / b = a / b - 1 := by
+  simpa only [← @div_self _ _ b h] using (div_sub_div_same a b b).symm
+
+theorem div_sub_one {a b : K} (h : b ≠ 0) : a / b - 1 = (a - b) / b :=
+  (div_sub_same h).symm
+
+theorem sub_div (a b c : K) : (a - b) / c = a / c - b / c :=
+  (div_sub_div_same _ _ _).symm
+
+/-- See `inv_sub_inv` for the more convenient version when `K` is commutative. -/
+theorem inv_sub_inv' {a b : K} (ha : a ≠ 0) (hb : b ≠ 0) : a⁻¹ - b⁻¹ = a⁻¹ * (b - a) * b⁻¹ := by
+  rw [mul_sub, sub_mul, mul_inv_cancel_right₀ hb, inv_mul_cancel ha, one_mul]
+
+theorem one_div_mul_sub_mul_one_div_eq_one_div_add_one_div (ha : a ≠ 0) (hb : b ≠ 0) :
+    1 / a * (b - a) * (1 / b) = 1 / a - 1 / b := by
+  rw [mul_sub_left_distrib (1 / a), one_div_mul_cancel ha, mul_sub_right_distrib, one_mul,
+    mul_assoc, mul_one_div_cancel hb, mul_one]
+
+-- see Note [lower instance priority]
+instance (priority := 100) DivisionRing.IsDomain : IsDomain K :=
```

Comment:
Is the name here correct? Am I supposed to add `inst` or maybe to use the auto generated one?